pennylane/operation.py

```diff
@@ -1866,12 +1866,14 @@ def __matmul__(self, other):
         else:
             raise ValueError("Can only perform tensor products between observables.")
 
-        if qml.QueuingContext.recording():
-            owning_info = qml.QueuingContext.get_info(self)["owns"] + (other,)
-
-            # update the annotated queue information
-            qml.QueuingContext.safe_update_info(self, owns=owning_info)
-            qml.QueuingContext.safe_update_info(other, owner=self)
+        if (
+            qml.QueuingContext.recording()
+            and self not in qml.QueuingContext.active_context()._queue
+        ):
+            qml.QueuingContext.append(self)
+
+        qml.QueuingContext.safe_update_info(self, owns=self.obs)
```

Comment:
`self.obs` may have to be put into a tuple, as one of the tests is failing:

https://github.com/PennyLaneAI/pennylane/runs/6782809546?check_suite_focus=true#step:11:149